spaceinvaders.py

```diff
@@ -1,648 +1,687 @@
+#!/usr/bin/env python
+
 # Space Invaders
 # Created by Lee Robinson
 
-#!/usr/bin/env python
 from pygame import *
 import sys
-from random import shuffle, randrange, choice
+from os.path import abspath, dirname
+from random import randrange, choice
+
+BASE_PATH = abspath(dirname(__file__))
+FONT_PATH = BASE_PATH + "/fonts/"
+IMAGE_PATH = BASE_PATH + "/images/"
+SOUND_PATH = BASE_PATH + "/sounds/"
 
 #           R    G    B
-WHITE 	= (255, 255, 255)
-GREEN 	= (78, 255, 87)
-YELLOW 	= (241, 255, 0)
-BLUE 	= (80, 255, 239)
-PURPLE 	= (203, 0, 255)
-RED 	= (237, 28, 36)
-
-SCREEN 		= display.set_mode((800,600))
-FONT = "fonts/space_invaders.ttf"
-IMG_NAMES 	= ["ship", "ship", "mystery", "enemy1_1", "enemy1_2", "enemy2_1", "enemy2_2",
-				"enemy3_1", "enemy3_2", "explosionblue", "explosiongreen", "explosionpurple", "laser", "enemylaser"]
-IMAGES 		= {name: image.load("images/{}.png".format(name)).convert_alpha()
-				for name in IMG_NAMES}
+WHITE = (255, 255, 255)
+GREEN = (78, 255, 87)
+YELLOW = (241, 255, 0)
+BLUE = (80, 255, 239)
+PURPLE = (203, 0, 255)
+RED = (237, 28, 36)
+
+SCREEN = display.set_mode((800, 600))
+FONT = FONT_PATH + "space_invaders.ttf"
+IMG_NAMES = ["ship", "ship", "mystery",
```

Comment:
While you're switching from tabs -> spaces (which I agree looks much cleaner) could you standardize using single quotes instead of double quotes?

Reply:
'Reformat quotes, double->single' (fd4fee2)